documentation/src/main/asciidoc/upgrading/upgrading-content.adoc

```diff
@@ -6,6 +6,13 @@ From 10.0 we only provide a PooledConnectionFactory based upon the link:https://
 This means that it is no longer possible to utilise `c3p0.properties` and `hikari.properties` files to configure the pool,
 instead an agroal compatible properties file can be provided. See link:../user_guide/user_guide.html#jdbc_cache_store[Connection Management] for more details.
 
+=== Cache and Cache Manager Listeners can now be configured to be non blocking
+Listeners in the past that were sync, were always ran in the thread that caused the event. Now these sync listeners
```

Comment:
s/were//

TBH I'm not sure if not blocking the caller thread is a worthy goal in itself... users that add listeners should be aware that their listeners will block the caller thread, and they are perfectly able to use a thread pool internally if they can't avoid blocking.

Reply:
They can use a thread pool internally, but they can't have it still block the operation until it is complete. This provides another option in their toolbelt.

Reply:
I don't mean the ability to return `CompletionStage<Void>` so that the caller thread doesn't block. I mean us assuming that all listeners that don't return `CompletionStage<Void>` are blocking and running them in a separate thread, when our documentation explicitly says "This means that you _must_ write your listener such that it does not block or do anything that takes too long".